edittypes/node_differ.py

```diff
@@ -215,48 +222,46 @@ def get_diff_count(result):
+
     for i in result['insert']:
-        sections_affected.add(i["section"])
-    for c in result['change']:
-        sections_affected.add(c['prev']["section"])
+        text = i['text']
+        is_edit_type_found,wikitext,edit_type = is_edit_type(text,i['type'])
+        #check if edit_type in edit types dictionary
+        if is_edit_type_found:
+            if edit_types.get(edit_type,{}):
+                edit_types[edit_type]['insert'] = edit_types[edit_type].get('insert', 0) + 1
+            else:
+                edit_types[edit_type] = {'insert':1}
 
-    edit_types = {}
-    for s in sections_affected:
-        for r in result['remove']:
-            if r["section"] == s:
-                text = r['text']
-                is_edit_type_found,wikitext,edit_type = is_edit_type(text,r['type'])
-                if is_edit_type_found:
-                    if edit_types.get(edit_type,{}):
-                        edit_types[edit_type]['remove'] = edit_types[edit_type].get('remove', 0) + 1
-                    else:
-                        edit_types[edit_type] = {'remove':1}
-
-        for i in result['insert']:
-            if i["section"] == s:
-                text = i['text']
-                is_edit_type_found,wikitext,edit_type = is_edit_type(text,i['type'])
-                #check if edit_type in edit types dictionary
-                if is_edit_type_found:
-                    if edit_types.get(edit_type,{}):
-                        edit_types[edit_type]['insert'] = edit_types[edit_type].get('insert', 0) + 1
-                    else:
-                        edit_types[edit_type] = {'insert':1}
-
-        for c in result['change']:
-            
-            if c["prev"]["section"] == s:
-                if c['prev']['type'] == c['curr']['type']:
-                    is_edit_type_found,edit_type = is_change_in_edit_type(c['prev']['text'],c['curr']['text'],c['prev']['type'])
-                    
-                    #check if edit_type in edit types dictionary
-                    if is_edit_type_found:
-                        if edit_types.get(edit_type,{}):
-                            edit_types[edit_type]['change'] = edit_types[edit_type].get('change', 0) + 1
-                        else:
-                            edit_types[edit_type] = {'change':1}
+    for c in result['change']:
+        if c['prev']['type'] == c['curr']['type']:
+            is_edit_type_found,edit_type = is_change_in_edit_type(c['prev']['text'],c['curr']['text'],c['prev']['type'])
+            #check if edit_type in edit types dictionary
+            if is_edit_type_found:
+                if edit_types.get(edit_type,{}):
+                    edit_types[edit_type]['change'] = edit_types[edit_type].get('change', 0) + 1
+                else:
+                    edit_types[edit_type] = {'change':1}
+
+    for m in result['move']:
+        text = m['text']        
+        is_edit_type_found,edit_type = is_edit_type(text, m['type'])
```

Comment:
I think this will throw an exception -- move dictionary has the `prev` and `curr` fields so probably `m['prev']['text']` and `m['prev']['type']`. Also `is_edit_type` returns three objects not two

Reply:
Alright